gamemodes/terrortown/gamemode/server/sv_weaponry.lua

```diff
@@ -666,6 +615,19 @@ function WEPS.DropNotifiedWeapon(ply, wep, deathDrop, keepSelection)
 	wep.IsDroppedBecauseDeath = nil
 end
 
+---
+-- Called whenever a @{Player} drops Ammo for a @{Weapon}
+-- @param Player ply The player whose ammo is about to be dropped
+-- @param Weapon wep The weapon to reference when dropping ammo
+-- @param boolean useClip If set to true the ammo is dropped from the clip, otherwise, from reserve ammo.
+-- @param number amt The amount of ammo to drop
+-- @realm server
+function WEPS.DropNotifiedAmmo(ply, wep, useClip, amt)
+	if not IsValid(ply) or not IsValid(wep) then return end
+
+	ply:DropAmmo(wep, useClip, amt)
+end
```

Comment:
This function does not contribute anything, does it? I think we should remove it

Reply:
I get why @EntranceJew added it. He wanted to reflect the way dropping weapons work. But I agree, it is confusing and adds no benefit. Especially since the current call-stack looks like this: `ply:SafeDropWeapon` -> `WEPS.DropNotifiedAmmo` -> `ply:DropWeapon`

Which is honestly kinda weird. But I guess it is done in the same way for weapon drops.

Reply:
Yes i feel the same way but there is no reason to do it this way just because the weapon drop also calls a function here